### PR TITLE
[v4] add `contentDirBasePath` option to configure `content` directory under a catch-all subdirectory

### PR DESCRIPTION
**Pull request by dimaMachina**
fixes https://github.com/shuding/nextra/issues/3567 https://github.com/shuding/nextra/discussions/3582

### CODE REVIEW

**Comment by Firgrep** [comment]
Question: will the content folder need to mimic the nested path from /app? E.g.: 
/App/mine/docs/[[...modxPaths]]/page.tsx
/Content/mine/docs/my-docs.mdx

Or can the content be anything?
/Content/my-docs.mdx

How will all of this relate to the items in the navigation if we use the basepath?

Thanks!

**Comment by dimaMachina** [comment]
> Question: will the content folder need to mimic the nested path from /app? E.g.:

nope, you don't need to mimic, you can see this docs example https://github.com/shuding/nextra/tree/v4-50/examples/docs

Also, I am thinking of removing `useContentDir` option since we can know whether `content` directory is used based on `catchAllBasePath` value, wdyt?

**Comment by Firgrep** [comment]
Alright, cool!

Yes, that makes sense, the user will have to set some value in the config anyhow to activate catch-all, so the ’useContentDir’ seems redundant.

File: .changeset/olive-seas-clap.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'nextra-theme-blog': patch
+'nextra-theme-docs': patch
+'nextra': patch
```
Comment (dimaMachina) [inline .changeset/olive-seas-clap.md:4]:
```suggestion
'nextra-theme-blog': feat
'nextra-theme-docs': feat
'nextra': feat
```

File: packages/nextra/src/server/loader.ts [hunk 2]
```diff
@@ -112,16 +112,19 @@ export async function loader(
...
-      dir: useContentDir ? path.join('content', locale) : APP_DIR,
-      isAppDir: !useContentDir
+      dir: contentDirBasePath ? path.join('content', locale) : APP_DIR,
+      isAppDir: !contentDirBasePath
     })
-
-    const { pageMap, mdxPages } = generatePageMapFromFilepaths(relativePaths)
+    const { pageMap, mdxPages } = generatePageMapFromFilepaths(
+      relativePaths,
+      // replace forward slash
```
Comment (dimaMachina) [inline packages/nextra/src/server/loader.ts:120]:
```suggestion
      // remove forward slash
```

**Comment by dimaMachina** [comment]
@Firgrep I found one issue for `docs` folder:

So now, this `_meta` is ignored from adding to `pageMap`

image

image

- we cannot rename `docs` folder here
- we cannot also use `display: 'children'` (to hide the folder and show only children).

This will be not fixed in this PR, and more likely be fixed after 4 RC.0.

When you set `contentDirBasePath: '/docs'` - pages `page.{tsx,mdx,jsx}` and `_meta` files  in `app` dir  are not added to `pageMap`, to fix it I have an idea: when `contentDirBasePath` is set - still add all `page.{tsx,mdx,jsx}` and `_meta` files from `app` dir to `pageMap`

**Comment by dimaMachina** [comment]
@Firgrep ok this is fixed 😁 now we have we have 2 modes working together `content` and `page.{jsx,tsx,mdx}` routes added both to `pageMap`  as well `_meta` from `content` and `_meta` from `app`

**Comment by Firgrep** [comment]
@dimaMachina Fixed as in you fixed what you said couldn't be fixed in this PR in the previous comment? I'm just reading it now. Ah, so there can co-exist a `_meta` file inside `/app` alongside `/content` ?

**Comment by dimaMachina** [comment]
fix coming in another PR today

**Comment by Firgrep** [comment]
This man is on 🔥 Can't wait for the next pre-release that includes these changes to implement it!

**Comment by dimaMachina** [comment]
See the video demonstrating the working behaviour of both modes together and they play well, it's for a whiiiile I had in mind at some point to make them together 😜


https://github.com/user-attachments/assets/038118dc-22ac-42d3-badc-65c87a0cb40e

**Comment by Firgrep** [comment]
@dimaMachina That's really awesome! Nextra is going to be more powerful than ever! You could even put that video up on the docs, or at least in your blog post.

**Comment by dimaMachina** [comment]
looks nice idea!

**Pull request merged**